test/test_linalg.py

```
@@ -8188,6 +8188,27 @@ def test_tensordot(self, device):
         an = torch.from_numpy(np.tensordot(np.zeros((), dtype=np.float32), np.zeros((), dtype=np.float32), 0))
         self.assertEqual(a, an)
 
+    @onlyCUDA
+    @skipCUDAIfNoMagma
+    @skipCUDAIfNoCusolver
+    def test_preferred_linalg_library(self):
+        # The main purpose of this test is to make sure these "backend" calls work normally without raising exceptions.
+        x = torch.randint(2, 5, (2, 4, 4), device='cuda', dtype=torch.double)
+
+        torch.backends.cuda.preferred_linalg_library('cusolver')
```

Is this sending a warning? If so, it can be nice to catch it to avoid spam.
Also we have something to make warn once always want and you can actually check that you do get the warning.

---

It's using TORCH_WARN_ONCE in Context.cpp. It raises `Warning` instead of `UserWarning` and doesn't work well with `assertWarnsOnceRegex`.

```
[W Context.cpp:157] Warning: torch.backends.cuda.preferred_linalg_library is an experimental feature. If you see any error or regression when this flag is set please file an issue on GitHub. (function operator())
```